docs/0.4/examples/gameroom/walkthrough/index.md

```diff
@@ -388,3 +422,970 @@ GET /api/proposals
+   admin service for a specific circuit management type (also described in
+   ​The Prequel, section P.3​).
+
+2. If there are any registered application authorization handlers for the
+   proposed circuit management type, each admin service forwards the request
+   to the local connected Gameroom application authorization handler.
+
+   The notification is sent on a WebSocket connection.
+
+``` json
+{
+"eventType": "ProposalSubmitted",
+"message": {
+  "proposal_type": "Create",
+  "circuit_id": "gameroom::acme-node-000::bubba-node-000::<UUIDv4>",
+  "circuit_hash": "...",
+  "circuit": {
+    "circuit_id":"gameroom::acme-node-000::bubba-node-000::<UUIDv4>",
+    "authorization_type": "Trust",
+    "members": [{
+      "node_id": "acme-node-000",
+      "endpoint": "tls://splinterd-node-acme:8044"
+      },
+    {
+      "node_id": "bubba-node-000",
+      "endpoint": "tls://splinterd-node-bubba:8044"
+      }
+    ],
+    "roster": [{
+      "service_id": "gameroom_acme-node-000",
+      "service_type": "scabbard",
+      "allowed_nodes": [ "acme-node-000" ],
+      "arguments": {
+        "peer_services": [ "gameroom_bubba-node-000" ],
+        "admin_keys": [
+          <acme gameroomd public key>
+          ]
+        }
+      },
+      {
+        "service_id": "gameroom_bubba-node-000",
+        "service_type": "scabbard",
+        "allowed_nodes": [ "bubba-node-000" ],
+        "arguments": {
+          "peer_services": [ "gameroom_acme-node-000" ],
+          "admin_keys": [
+            <acme gameroomd public key>
+            ]
+          }
+        }
+      ],
+      "circuit_management_type": "gameroom",
+      "application_metadata":​ <metadata bytes defined by the application>​,
+      "persistence": "Any",
+      "durability": "None",
+      "routes": "Any",
+    },
+    "vote_record": [{}],
+    "requester": "public_key_of_requester",
+    "requester_node_id": "acme-node-000"
+  }
+}
+```
+
+#### I-2.8. Gameroom daemons write notification to Gameroom database
+
+When each Gameroom application authorization handler receives the gameroom
+proposal on the WebSocket connection, it parses the information and adds it to
+several tables in the Gameroom daemon's local database: `​gameroom​`, ​
+`gameroom_member​`, `​gameroom_service​`, `gameroom_proposal​`, and
+`​gameroom_notification​`.
+
+##### I-2.8.1. New `gameroom` table entry
+
+First, the Gameroom application authorization handler adds a new entry to the
+​gameroom​ table. This table contains the information about the circuit
+definition, including the data that was passed in the ​application_metadata​ field.
+
+``` sql
+CREATE TABLE IF NOT EXISTS  gameroom (
+  circuit_id TEXT PRIMARY KEY,
+  authorization_type TEXT NOT NULL,
+  persistence TEXT NOT NULL,
+  routes TEXT NOT NULL,
+  durability TEXT NOT NULL,
+  circuit_management_type TEXT NOT NULL,
+  alias TEXT NOT NULL,
+  status TEXT NOT NULL,
+  created_time TIMESTAMP NOT NULL,
+  updated_time TIMESTAMP NOT NULL,
+);
+```
```

Comment:
The original version used spaces to align the TEXT and PRIMARY KEY/NOT NULL "columns".  Is the spacing significant?

Reply:
Nope